src/server/routes/register.js

```diff
@@ -0,0 +1,26 @@
+const express = require('express')
+const router = express.Router()
+const UserModel = require('../models/user')
+const logger = require('../logger.js')
+const Utilities = require('../utilities')
+
+const saltSize = 70
+
+router.post('/', (req, res) => {
+
+	UserModel.on('index', err => logger.error(`Not created indexes: ${err}`))
+	
+	const userInstance = new UserModel(req.body)
+
+	userInstance.salt = Utilities.generateSalt(saltSize)
```

Comment:
Let's just generate hashes before creating userInstance and pass everything to the constructor.
Also we don't have any place where we place info about what kind of info do we expect from req.body.
```

const salt = Utilities.generateSalt(saltSize)
const password = Utilities.hashPassword(userInstance.salt, userInstance.password)
const userData = {
    username: req.body.username,
    email: req.body.email,
    salt,
    password
}
const userInstance = new UserModel(userData)
...

```